orchestrator/main_loops.go

```diff
@@ -314,10 +371,35 @@ func (p *gravityOrchestrator) BatchRequesterLoop(ctx context.Context) (err error
 					return nil
 				}
 
-				logger.Info().Str("token_contract", tokenAddr.String()).Str("denom", denom).Msg("sending batch request")
+				shouldRequestBatch := true
+
+				if p.relayer.GetProfitMultiplier() != 0.0 {
+					// First we get the cost of the transaction in USD
+					totalETHcost := big.NewInt(0).Mul(gasPrice, big.NewInt(estimatedGasCosts[unbatchedToken.TxCount]))
```

Comment:
Do we need to do a bounds check?

Reply:
We don't need to, as it's guaranteed by the chain that the max amount of txs will be 100. That being said, I just realize there's a bug here lol (should be `TxCount - 1`). Will fix